### PR TITLE
[dy] Add block execution logic

### PR DESCRIPTION
**Pull request by dy46**
# Summary

Add logic to execute block code.

# Tests

unit tests

cc: @wangxiaoyou1993

### CODE REVIEW

File: mage_ai/data_preparation/models/block.py [hunk 1]
```diff
@@ -68,7 +75,7 @@ def create(self, name, block_type, repo_path):
             pass
         with open(os.path.join(block_dir_path, f'{uuid}.py'), 'w'):
             pass
-        return Block(name, uuid, block_type)
```
Comment (wangxiaoyou1993) [inline mage_ai/data_preparation/models/block.py:71]: If pipeline is not None, let's call `pipeline.add_block(block)` here.

File: mage_ai/tests/data_preparation/models/test_block.py [hunk 2]
```diff
@@ -29,6 +32,88 @@ def test_create(self):
...
+def load_data():
+    data = {'col1': [1, 1, 3], 'col2': [2, 2, 4]}
+    df = pd.DataFrame(data)
+    return [df]
+            ''')
+        with open(block2.file_path, 'w') as file:
+            file.write('''import pandas as pd
+@transformer
+def remove_duplicate_rows(df):
+	df_transformed = df.drop_duplicates()
```
Comment (wangxiaoyou1993) [inline mage_ai/tests/data_preparation/models/test_block.py:53]: nit: use 4 spaces indentation

File: mage_ai/tests/data_preparation/models/test_block.py [hunk 3]
```diff
@@ -29,6 +32,88 @@ def test_create(self):
...
+        asyncio.run(block2.execute())
+
+        variable_manager = VariableManager(pipeline.repo_path)
+        data = variable_manager.get_variable(
+            pipeline.uuid,
+            block2.uuid,
+            'df',
+            variable_type='dataframe'
+        )
+        self.assertEqual(len(data.index), 2)
```
Comment (wangxiaoyou1993) [inline mage_ai/tests/data_preparation/models/test_block.py:68]: You can use 
```
from pandas.util.testing import assert_frame_equal
assert_frame_equal(df1, df2)
```

File: mage_ai/tests/data_preparation/models/test_block.py [hunk 4]
```diff
@@ -29,6 +32,88 @@ def test_create(self):
...
+def load_data():
+    data = {'col1': [5], 'col2': [6]}
+    df = pd.DataFrame(data)
+    return [df]
+            ''')
+        with open(block3.file_path, 'w') as file:
+            file.write('''import pandas as pd
+@transformer
+def union_datasets(df1, df2):
+	df_union = pd.concat([df1, df2])
```
Comment (wangxiaoyou1993) [inline mage_ai/tests/data_preparation/models/test_block.py:98]: nit: 4 spaces indentation

File: mage_ai/tests/data_preparation/models/test_block.py [hunk 5]
```diff
@@ -29,6 +32,88 @@ def test_create(self):
...
+        asyncio.run(block3.execute())
+
+        variable_manager = VariableManager(pipeline.repo_path)
+        data = variable_manager.get_variable(
+            pipeline.uuid,
+            block3.uuid,
+            'df',
+            variable_type='dataframe'
+        )
+        self.assertEqual(len(data.index), 3)
```
Comment (wangxiaoyou1993) [inline mage_ai/tests/data_preparation/models/test_block.py:115]: same here about `assert_frame_equal`

File: mage_ai/tests/data_preparation/models/test_block.py (hunk 2 above)
Comment (dy46) [reply mage_ai/tests/data_preparation/models/test_block.py:53]: i'm not sure why it looks like this in the PR, but it only shows a 4 space indent in my editor

**Comment by dy46** [comment]
@wangxiaoyou1993 updated, PTAL

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**